sdk/opendp/whitenoise/ast/validate.py

```diff
@@ -37,13 +37,46 @@ def validate_all(self):
 
         checks = [func for func in dir(QueryConstraints) if callable(getattr(QueryConstraints, func)) and func.startswith("check_")]
         rets = [getattr(self, check)() for check in checks]
-
+            
     def check_aggregate(self):
-        nes = self.query.select.namedExpressions
-        agg = self.query.agg
-        gc = agg.groupedColumns() if agg is not None else []
-        exp = [c.expression for c in nes]
-        agg = [e for e in exp if type(e) == AggFunction and e.is_aggregate()]
+        q = self.query
+        def simplify(exp):
+            while type(exp) is NestedExpression:
+                exp = exp.expression
+            if type(exp) is ArithmeticExpression:
+                l = simplify(exp.left)
+                r = simplify(exp.right)
+                if type(l) is Literal:
+                    exp = r
+                elif type(r) is Literal:
+                    exp = l
+            if isinstance(exp, (PowerFunction, RoundFunction)):
+                exp = exp.expression
+            return exp
+
+        select_expressions = [simplify(s[1]) for s in q.m_symbols]
+        is_agg = lambda n: type(n) == AggFunction and n.is_aggregate
+        split = lambda p, i: (list(filter(p, i)), list(filter(lambda v: not p(v), i)))
```

Comment:
I think python supports this better with list comprehensions. Although the functional approach is very clean and I was able to follow it easily. filter in python is inferior to list comprehensions since it requires the lambda for not where you can do the below code with list comprehensions and remove the need for a split function.

aggs = [val if is_agg(val) for val in select_expressions]
non_aggs = [val if not is_agg(val) for val in select_expressions]

Reply:
Good call.  In functional languages, it's a single-line command that splits an iterable based on a single predicate, and it makes me nervous that python forces me to copy the predicate reference in two places.  But my "fix" is worse than the problem.  I'll change it.